src/radical/entk/utils/init_transition.py

```diff
@@ -28,9 +28,14 @@ def transition(obj, obj_type, new_state, channel, queue, profiler, logger):
 
         logger.info('Transition of %s to new state %s successful' % (obj.uid, new_state))
 
-    except Exception, ex:
 
-        logger.exception('Transition of %s to %s state failed, error: %s' % (obj.uid, new_state, ex))
+    except Exception:
+
+        logger.exception('Transition of %s to %s state failed,',
+                         obj.uid, new_state)
+
+        # AM: why the reset if the transition failed?  Or is that
+        # non-transactional after all?
```

Comment:
The try-except block together make the transition function transactional. 
sync_with_master() is not transactional. 

Does that help? Maybe I am missing something in the question.
Let me know.

Reply:
@vivek-bala ping?

Reply:
@vivek-bala: ping?